.paw/work/workflow-handoffs/ImplementationPlan.md

```diff
@@ -0,0 +1,669 @@
+- Update package.json `contributes.languageModelTools` array with new tool definition
+
+**File**: `package.json`
+**Changes**:
+- Add tool definition to `contributes.languageModelTools` array:
+  - `name`: `paw_call_agent`
+  - `displayName`: `Call PAW Agent`
+  - `modelDescription`: Tool description explaining handoff functionality
+  - `inputSchema`: JSON schema with `target_agent`, `work_id`, `inline_instruction` properties
+
+#### 3. WorkflowContext.md Field Addition
+**File**: `src/prompts/workflowInitPrompt.ts`
+**Changes**:
+- Add `HANDOFF_MODE` variable to template: `Handoff Mode: ${HANDOFF_MODE}`
+- Default to "manual" mode
+- Add conditional logic: if Auto mode selected, enforce local review strategy (reject prs strategy with error)
+
+**File**: `src/ui/userInput.ts`
+**Changes**:
+- Add `HandoffMode` type: `'manual' | 'semi-auto' | 'auto'`
+- Add `collectHandoffMode()` function returning QuickPick with options:
+  - Manual: "Full control - you command each stage transition"
+  - Semi-Auto: "Thoughtful automation - automatic at research/review, pause at decisions"
+  - Auto: "Full automation - agents chain through all stages (local strategy required)"
+- Add validation: if Auto mode + prs strategy selected, show error and reset to local strategy
+
+#### 4. Documentation Updates
+**File**: `README.md`
+**Changes**:
+- Add "Workflow Handoffs" section explaining Manual/Semi-Auto/Auto modes
+- Add examples of handoff commands (`research`, `implement Phase 2`, `status`)
+- Document inline instruction syntax ("continue Phase 2 but add rate limiting")
+
+### Success Criteria:
+
+#### Automated Verification:
+- [ ] Unit tests pass for `handoffTool.ts`: `npm test`
+- [ ] TypeScript compilation succeeds: `npm run compile`
+- [ ] Tool registered and discoverable in Extension Development Host
+- [ ] Validation tests pass: missing ImplementationPlan.md throws error, invalid Work ID rejected
+- [ ] Linting passes: `npm run lint`
+
+#### Manual Verification:
+- [ ] Extension activates without errors
+- [ ] Handoff tool appears in tool approval UI when agent invokes
+- [ ] Manual mode: Typing `research` after Spec completion opens Spec Research Agent chat
+- [ ] Work ID correctly passed and `paw_get_context` succeeds in target agent
+- [ ] Inline instruction ("continue but add logging") appears in target agent's initial context
+- [ ] Prerequisite validation: `implement Phase 1` without ImplementationPlan.md shows error message
+- [ ] Agent name mapping: `implement Phase 2` resolves to `PAW-03A Implementer`
+
+---
+
+## Phase 2: Status Agent Core Enhancements
+
+### Overview
+Transform the Status Agent into a comprehensive workflow navigation hub by adding directory scanning for active workflows, artifact detection, git status checking, PR querying, and actionable next-step suggestions.
+
+### Changes Required:
+
+#### 1. Workflow Scanning Logic
+**File**: `src/tools/statusTool.ts` (new)
```

Comment:
We do not need a status tool.

The agent instructions should tell the agent what to check where in order to figure out status. It does not need a procedural tool to hardcode how to look for status. Let the agent use its available tools and logic to determine the status. This should all be prompt driven and not require any new tools for status discovery.